disnake/ext/commands/params.py

```diff
@@ -180,18 +182,86 @@ def _xt_to_xe(xe: Optional[float], xt: Optional[float], direction: float = 1) ->
 
 
 class Injection:
+    """Represents a slash command injection
+
+    .. versionadded:: 2.3
+
+    .. versionchanged:: 2.6
+        Constructor now accept keyword-only argument ``autocompleters``
+
+    Attributes
+    ----------
+    function: Callable
+        The underlying injection
+    """
+
     _registered: ClassVar[Dict[Any, Injection]] = {}
 
     function: Callable
+    autocompleters: Dict[str, Callable]
+
+    def __init__(self, function: Callable, *, autocompleters: Dict[str, Callable] = ...) -> None:
+        if not asyncio.iscoroutinefunction(function):
+            raise TypeError("Injection must be a function")
+
+        if autocompleters is not Ellipsis and any(
+            [not asyncio.iscoroutinefunction(f) for f in autocompleters.values()]
+        ):
+            raise TypeError("Some of your autocompleters are not coroutines")
 
-    def __init__(self, function: Callable) -> None:
         self.function = function
+        self.autocompleters = autocompleters if autocompleters is not Ellipsis else {}
 
     @classmethod
-    def register(cls, function: CallableT, annotation: Any) -> CallableT:
+    def register(
```

Comment:
missing docstring and `.. versionchanged` for the return type change

Reply:
Should we really expose `@Injection.register()` (i.e., by adding docstring Sphinx will automatically add `@register` to docs because there's `.. autoclass`, which means `@register` is allowed to be used by user)?

Reply:
Ah, good point. Let's leave it as is then.